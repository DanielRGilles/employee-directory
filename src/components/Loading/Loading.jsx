import React from 'react'
import loading from '../../assets/loading.png'

export default function Loading() {
    return (
        <div>
            <img src={loading} alt='loading spinner' />
            <h1 className='loading'>......Loading</h1>
        </div>
    )
}
