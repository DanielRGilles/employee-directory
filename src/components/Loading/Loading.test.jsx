import {render} from  '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { UserProvider } from '../../context/UserContext';
import Loading from './Loading';

jest.mock('../../context/UserContext')
jest.mock('../../services/users.js')

describe('Loading component test', () => {
  
    it('renders the Loading spinner', () => {
       const { container } = render(
            <MemoryRouter>
                <UserProvider>
                     <Loading/>
                </UserProvider>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
         
    })

})