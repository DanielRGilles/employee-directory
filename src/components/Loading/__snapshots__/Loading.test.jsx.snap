// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component test renders the Loading spinner 1`] = `
<div>
  <div>
    <img
      alt="loading spinner"
      src="/path/image"
    />
    <h1
      class="loading"
    >
      ......Loading
    </h1>
  </div>
</div>
`;
